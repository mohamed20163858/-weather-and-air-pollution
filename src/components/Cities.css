.search {
  width: 100%;
  background-color: #5384e0;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px;
  box-sizing: border-box;
}

.search input {
  background-color: #5384e0;
  color: white;
  padding-left: 20px;
}

.search input::placeholder {
  padding-left: 10px;
  color: white;
}

.search i {
  color: white;
}

.cards {
  display: flex;
  flex-wrap: wrap;
}

.city-card {
  width: 50%;
  border: none;
  text-decoration: none;
}

.city-card:nth-child(2n+1) {
  background-color: #4066ad;
  color: white;
}

.city-card:nth-child(2n) {
  background-color: #3e61a3;
  color: white;
}
