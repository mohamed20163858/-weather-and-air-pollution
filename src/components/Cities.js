import { useSelector } from 'react-redux';
import { useState } from 'react';
import CityCard from './cityCard';
import './Cities.css';

const Cities = () => {
  const infoUI = [];
  const infoStore = useSelector((store) => store.citiesInfo);
  const [infoState, setInfoState] = useState([]);
  const [flag, setFlag] = useState(true);
  const limit = 101;
  const writingAction = (e) => {
    let shownData;
    if (e.target.value !== '') {
      shownData = infoStore.filter(
        (element) => element.name.trim().toLowerCase().startsWith(e.target.value.toLowerCase()),
      );
      setFlag(false);
    } else {
      shownData = infoStore;
      setFlag(true);
    }
    setInfoState(shownData);
  };
  if (infoState.length === 0 && flag) {
    infoState.push(...infoStore);
  }
  for (let i = 0; i < infoState.length; i += 1) {
    infoUI.push(<CityCard
      countryName={infoState[i].name}
      cityName={infoState[i].capital}
      capitalISO2={infoState[i].iso2}
      capitalISO3={infoState[i].iso3}
      id={i}
      key={`city-${i}`}
    />);
    if (i === limit) {
      break;
    }
  }
  if (infoUI.length === 0) {
    infoUI.push(<p key="unvalid-data">there is no country has this name please enter a valid country name</p>);
  }
  return (
    <div className="cards" data-testid="momo-id">
      <div className="search">
        <i className="fa-solid fa-magnifying-glass" />
        <input type="text" placeholder="search" onKeyUp={writingAction} />
      </div>
      {infoUI}
    </div>
  );
};
export default Cities;
