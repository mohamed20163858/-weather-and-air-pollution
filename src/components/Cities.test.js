import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import '@testing-library/jest-dom';
import Cities from './Cities';

function TestCities() {
  return (
    <Provider store={store}>
      <Router>
        <Cities />
      </Router>
    </Provider>
  );
}
it('renders correctly', () => {
  const tree = renderer
    .create(<TestCities />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
