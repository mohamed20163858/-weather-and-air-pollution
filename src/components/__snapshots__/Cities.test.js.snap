// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cards"
  data-testid="momo-id"
>
  <div
    className="search"
  >
    <i
      className="fa-solid fa-magnifying-glass"
    />
    <input
      onKeyUp={[Function]}
      placeholder="search using country, city"
      type="text"
    />
  </div>
  <p>
    there is no country has this name please enter a valid country name
  </p>
</div>
`;
